connector/azure/graph/src/main/java/com/blazebit/query/connector/azure/graph/UserDataFetcher.java

```diff
@@ -45,7 +48,13 @@ public List<User> fetch(DataFetchContext context) {
             List<GraphServiceClient> graphServiceClients = AzureGraphConnectorConfig.GRAPH_SERVICE_CLIENT.getAll(context);
             List<User> list = new ArrayList<>();
             for (GraphServiceClient graphServiceClient : graphServiceClients) {
-                list.addAll(graphServiceClient.users().get().getValue());
+                List<String> servicePlanNames = getAllServicePlanNames(graphServiceClient.subscribedSkus().get().getValue());
+                if (servicePlanNames.contains("AAD_PREMIUM") || servicePlanNames.contains("AAD_PREMIUM_P2")) {
```

Comment:
I also suggest adding an enum for these values. Maybe there is already one provided by the sdk?

Reply:
done